### PR TITLE
[Assignment4] dockerize user and ques service

### PR DESCRIPTION
**Pull request by LeirdaKJW**
## Dockerize user service and question service
This PR will introduce docker support for both user and question service, additionally, this completes assignment 4's requirements.
**Checklist:**
- [x] Containerize question repository application.
- [x] Containerize the user profile management application.
**Changes Made:**
- Added Docker Compose configuration for local development.
- Created Dockerfiles for ques and user service.
- Defined environment variables and dependencies in the Docker Compose files.
- Provided instructions for building and running the project in Docker containers.

### CODE REVIEW

File: backend/user-service/src/db/dbConfig.ts [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 });
+*/
+
+const sequelize = new Sequelize({
+  database: process.env.PG_DB,
+  username: process.env.PG_USER,
+  password: process.env.PG_PASSWORD,
+  host: process.env.PG_HOST, // This should match the service name in your Docker Compose file
+  dialect: 'postgres',
+});
```
Comment (dsja612) [inline backend/user-service/src/db/dbConfig.ts]: Did some testing:
- This localises the postgres db instance, meaning we can eliminate the need for elephantDB's cloud instance.
- Data also persists in the docker image itself.

GJ 👍

File: dockerfiles/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+1. Edit docker-compose.yml file
+2. To connect to mongodb questiondb for question-service, edit the following env variable in the file:
+- `MONGOURL`: mongodb connection string
+3. Run docker-compose up
+
+# setup for cloud postgres user-db:
+1. To connect to elephantsql user database, edit the following env variable in the file:
+- `POSTGRESURL`: postgres connection string
+2. Edit dbConfig file in backend/user-service/src/db so that sequelize uses the connection string.
+3. Run docker-compose up
```
Comment (dsja612) [inline dockerfiles/README.md]: Prob need to rewrite this

**Review by dsja612 — CHANGES_REQUESTED** [review]
LGTM after fixing readme

We might not need elephantSQL anymore, so we should fix the readme accordingly

File: dockerfiles/README.md (hunk 2 above)
Comment (LeirdaKJW) [reply dockerfiles/README.md]: Resolved in the following commits

**Review by dsja612 — APPROVED** [review]
lgtm lets merge this after assignment 3

**Pull request merged**